lib/improver/wind_gust_diagnostic.py

```diff
@@ -0,0 +1,232 @@
+        self.percentile_gust = percentile_gust
+        self.percentile_windspeed = percentile_windspeed
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        desc = ('<WindGustDiagnostic: wind-gust perc='
+                '{0:3.1f}, wind-speed perc={1:3.1f}>'
+                .format(self.percentile_gust, self.percentile_windspeed))
+        return desc
+
+    def add_metadata(self, cube):
+        """Add metadata to cube for windgust diagnostic.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.standard_name = "wind_speed_of_gust"
+        result.long_name = "wind_gust_diagnostic"
+        if self.percentile_gust == 50.0 and self.percentile_windspeed == 95.0:
+            diagnostic_txt = 'Typical gusts'
+        elif (self.percentile_gust == 95.0 and
+              self.percentile_windspeed == 100.0):
+            diagnostic_txt = 'Extreme gusts'
+        else:
+            diagnostic_txt = str(self)
+        result.attributes.update({'wind_gust_diagnostic': diagnostic_txt})
+
+        return result
+
+    @staticmethod
+    def update_metadata_after_max(cube, perc_coord):
+        """Update metadata after MAX found through merged_cube.collapsed
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data.
+        Returns:
+            result (iris.cube.Cube):
+                Cube containing the wind-gust diagnostic data with
+                corrected Metadata.
+
+        """
+        result = cube
+        result.remove_coord(perc_coord)
+        return result
+
+    @staticmethod
+    def extract_percentile_data(cube, req_percentile, standard_name):
+        """Extract percentile data from cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube contain one or more percentiles of wind_gust data.
+            req_percentile (float):
+                Required percentile value
+            standard_name (str):
+                Standard name of the data.
+
+        Returns:
+            (tuple) : tuple containing:
+                **result** (iris.cube.Cube):
+                    Cube containing the required percentile data
+                **perc_coord** (iris.coords.Coord):
+                    Percentile coordinate.
+
+        """
+        if not isinstance(cube, iris.cube.Cube):
+            msg = ('Expecting {0:s} data to be an instance of '
+                   'iris.cube.Cube but is'
+                   ' {1:s}.'.format(standard_name, type(cube)))
+            raise TypeError(msg)
+        perc_coord = None
+        perc_found = 0
+        for coord in cube.coords():
+            if coord.name().find('percentile') >= 0:
```

Comment:
We need to start implementing attribute axes sooner rather than later to make this whole process of identifying percentile, probability etc axes much more straightforward.

Reply:
Good idea. Is there an issue for this? Can we do this in IMPROVER or does this require an Iris change?